sql/catalyst/src/main/scala/org/apache/spark/sql/connector/catalog/V1Table.scala

```diff
@@ -51,38 +52,36 @@ private[sql] case class V1Table(v1Table: CatalogTable) extends Table {
     }
   }
 
-  def catalogTable: CatalogTable = v1Table
-
   lazy val options: Map[String, String] = {
-    v1Table.storage.locationUri match {
+    catalogTable.storage.locationUri match {
       case Some(uri) =>
-        v1Table.storage.properties + ("path" -> uri.toString)
+        catalogTable.storage.properties + ("path" -> uri.toString)
       case _ =>
-        v1Table.storage.properties
+        catalogTable.storage.properties
     }
   }
 
-  override lazy val properties: util.Map[String, String] = v1Table.properties.asJava
+  override lazy val properties: util.Map[String, String] = catalogTable.properties.asJava
```

Comment:
I leave the options/properties unchanged here, but we need to figure it out later. Currently there are 2 directions:
1. We have table options and table properties. Table options are special table properties with "option." prefix in its name. Table options will be extracted and used as scan/write options.
2. We only have table properties. The OPTIONS clause in CREATE TABLE should be the same as the TBLPROPERTIES clause.

We can have more discussion about it later.

Reply:
V2 has only table properties, so I think the question is whether we want to mix options into those table properties directly, or whether we want to prefix them so they can be recovered. I'm in favor of being able to recover them so we can constract the catalog table as it would be in the v1 path.